File: AndroidBasics/CustomToastExa/app/src/main/java/com/mehru/customtoastexa/MainActivity.java
package com.mehru.customtoastexa;

import static android.widget.Toast.LENGTH_LONG;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.view.Gravity;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class MainActivity extends AppCompatActivity {

    Button btnShow ;

    @SuppressLint("MissingInflatedId")
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);

        btnShow = findViewById(R.id.btnShow);

           //Default Toast
        //Toast.makeText(this, "This is Mt Toast", Toast.LENGTH_LONG).show();

        //Creating Own Custom Toast


        btnShow.setOnClickListener(new View.OnClickListener() {

            @Override
            public void onClick(View v) {

                Toast toast = new Toast(getApplicationContext());
                View view = getLayoutInflater().inflate(R.layout.custom_toast_layout,findViewById(R.id.viewContainer));
                toast.setView(view);

                TextView txtMsg = view.findViewById(R.id.txtmsg);
                txtMsg.setText("message sent Succcesfully");
                toast.setDuration(Toast.LENGTH_SHORT);
                toast.setGravity(Gravity.TOP,0,0);
                toast.show();

            }
        });


//        Toast toast = new Toast(getApplicationContext());
//          View view = getLayoutInflater().inflate(R.layout.custom_toast_layout,findViewById(R.id.viewContainer));
//          toast.setView(view);
//
//        TextView txtMsg = view.findViewById(R.id.txtmsg);
//        txtMsg.setText("message sent Succcesfully");
//        toast.setDuration(Toast.LENGTH_LONG);
//        toast.show();







    }
}

File: app dev kt/customizedialogbox/app/src/main/java/com/mehru/customizedialogbox/MainActivity.kt
package com.mehru.customizedialogbox

import android.annotation.SuppressLint
import android.app.Dialog
import android.os.Bundle
import android.widget.Button
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat

class MainActivity : AppCompatActivity() {

    lateinit var dialog :Dialog
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        enableEdgeToEdge()
        setContentView(R.layout.activity_main)

        var btnBook = findViewById<Button>(R.id.btnBook)

        dialog = Dialog(this)
        dialog.setContentView(R.layout.dialog_box)
        dialog.window?.setBackgroundDrawable(getDrawable(R.drawable.bg_dialog))

        var btnOrder = dialog.findViewById<Button>(R.id.btnOrder)

        btnBook.setOnClickListener {
            dialog.show()
        }

        btnOrder.setOnClickListener {
            dialog.dismiss()
        }

    }
}